### DISCUSSION TITLE
onTouch / onClick loading

### DISCUSSION THREAD

**Discussion by 0gust1** [spences10/sveltekit-embed#502] [Ideas]
First, thanks for this lib, it's really nice and useful to have all those embeds packaged together!

For web-performance, data frugality and privacy purposes, could it be possible to have a loading that could be conditioned to the user's explicit tap/click action ?
 
The Observer feature is quite nice, but on an embed heavy page, and for a user with limited data/connection (it happens more often than we think), just scrolling the page would load all the iframes (and trigger a lot of data/files fetching behind the scenes.)

(As an example this embed component (youtube) does a good job to only initially load the thumbnail (instead of the whole iframe): https://github.com/sharu725/youtube-embed).

**Comment by spences10** [comment]
Hey @0gust1 that's an awesome suggestion, yes!

I made #488 a while back to look into adding additional functionality to the YouTube embed but this suggestion is great!

I'm thinking that this functionality could possibly be built into the observer but it's been a while since I've done anything with this project 🙃

If a 'tap to load' function can be added to the observer it will reduce a load of data being used on page load

If you want to make any contributions, further suggestions I'd love to see them

thanks

**Reply by 0gust1** [reply]
OMG, I forgot I had opened this discussion (slow clap to me).

I'll look into this soon (as I used your package on a recent project, I have the incentives now :D )
